Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ExposureKeyHistoryCalculations.kt

```diff
@@ -1,39 +1,70 @@
 package de.rki.coronawarnapp.submission
 
+import KeyExportFormat
+import androidx.annotation.VisibleForTesting
 import com.google.android.gms.nearby.exposurenotification.TemporaryExposureKey
 
 class ExposureKeyHistoryCalculations(
+    private val submissionStatusRepository: SubmissionStatusRepository,
     private val transmissionRiskVectorDeterminator: TransmissionRiskVectorDeterminator,
+    private val daysSinceOnsetOfSymptomsVectorDeterminator: DaysSinceOnsetOfSymptomsVectorDeterminator,
     private val keyConverter: KeyConverter
 ) {
 
+    companion object {
+        const val VECTOR_LENGTH = 15
+    }
+
     fun transformToKeyHistoryInExternalFormat(
         keys: List<TemporaryExposureKey>,
         symptoms: Symptoms
     ) =
         toExternalFormat(
-            toSortedHistory(limitKeyCount(keys)),
-            transmissionRiskVectorDeterminator.determine(symptoms)
+            limitKeyCount(toSortedHistory(keys)),
+            transmissionRiskVectorDeterminator.determine(symptoms),
+            daysSinceOnsetOfSymptomsVectorDeterminator.determine(symptoms, VECTOR_LENGTH)
         )
 
-    fun <T> limitKeyCount(keys: List<T>): List<T> =
-        keys.take(MAXIMUM_KEYS)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun limitKeyCount(keys: List<TemporaryExposureKey>) =
+        keys.filter { it.daysAgo <= 14 }
 
-    fun toExternalFormat(
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun toExternalFormat(
         keys: List<TemporaryExposureKey>,
-        transmissionRiskVector: TransmissionRiskVector
-    ) =
-        keys.mapIndexed { index, key ->
-            // The latest key we receive is from yesterday (i.e. 1 day ago),
-            // thus we need use index+1
-            keyConverter.toExternalFormat(key, transmissionRiskVector.getRiskValue(index + 1))
+        transmissionRiskVector: TransmissionRiskVector,
+        daysSinceOnsetOfSymptomsVector: DaysSinceOnsetOfSymptomsVector
+    ): List<KeyExportFormat.TemporaryExposureKey> {
+        val result = mutableListOf<KeyExportFormat.TemporaryExposureKey>()
+        keys.groupBy { it.daysAgo }.forEach { entry ->
+            val index = daysSinceOnsetOfSymptomsVector.indexOf(entry.key)
+            val today = entry.key == 0
```

Comment:
I find it quite confusing to understand what `entry.key` is referring to. At first, I was thinking it's the key of the TEK (which is actually stored in the `keyData` attribute) until I realized it's `daysAgo`.
For readability, I would suggest to add something like `val daysAgo = entry.key` as the first thing.